src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/tools/runInTerminalTool.ts

```diff
@@ -415,15 +422,15 @@ export class RunInTerminalTool extends Disposable implements IToolImpl {
 		}));
 
 		if (args.isBackground) {
-			let outputAndIdle: { terminalExecutionIdleBeforeTimeout: boolean; output: string; pollDurationMs?: number; modelOutputEvalResponse?: string } | undefined = undefined;
+			let outputAndIdle: IPollingResult & { pollDurationMs: number } | undefined = undefined;
 			let outputMonitor: OutputMonitor | undefined = undefined;
 			try {
 				this._logService.debug(`RunInTerminalTool: Starting background execution \`${command}\``);
 
 				const execution = new BackgroundTerminalExecution(toolTerminal.instance, xterm, command);
 				RunInTerminalTool._backgroundExecutions.set(termId, execution);
 
-				outputMonitor = this._instantiationService.createInstance(OutputMonitor, execution);
+				outputMonitor = this._instantiationService.createInstance(OutputMonitor, execution, this._languageModelsService, this._taskService, this._chatWidgetService, undefined);
```

Comment:
We want these to be injected via DI, then RunInTerminalTool doesn't need them in the ctor/imports too

Reply:
It's because you have an optional `?` at the end. All the injectable ones need to be last, so no `?` can be used.